File: 0817-design-hashmap/0817-design-hashmap.py
class _Node:
        def __init__(self, key = -1, data = -1, next=None):
            self.key = key
            self.data = data
            self.next = next

class MyHashMap:

    def __init__(self):
        self.nodes = [ _Node() for node in range(1000)]
        self.size = len(self.nodes)

    # def hash_function(key: int) -> int:
    #     return key % self.size

    def put(self, key: int, value: int) -> None:
        index = key % self.size
        head = self.nodes[index]
        while head.next:
            if head.next.key == key:
                head.next.data = value
                return
            head = head.next
        head.next = _Node(key=key, data=value)
        
    def get(self, key: int) -> int:
        index = key % self.size
        head = self.nodes[index].next
        while head:
            if head.key == key:
                return head.data
            head = head.next
        return -1

    def remove(self, key: int) -> None:
        index = key % self.size
        head = self.nodes[index]
        while head and head.next:
            if head.next.key == key:
                head.next = head.next.next
                return
            head = head.next


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)

File: 0817-design-hashmap/0817-design-hashmap.java
class MyHashMap {

    private static class Node {
        int key, value;
        Node next;

        public Node(int key, int value, Node next) {
            this.key = key;
            this.value = value;
            this.next = next;
        }

        public Node() {
            this(-1, -1, null);
        }
    }

    private final Node[] hashMap;

    public MyHashMap() {
        hashMap = new Node[10000];
        for (int i = 0; i < hashMap.length; i++) {
            hashMap[i] = new Node();
        }
    }
    
    private int generateHashKey(int key) {
        return key % hashMap.length;
    }

    public void put(int key, int value) {
        int hashNumber = generateHashKey(key);
        Node current = hashMap[hashNumber];

        while (current.next != null) {
            if (current.next.key == key) {
                current.next.value = value;
                return;
            }

            current = current.next;
        }

        current.next = new Node(key, value, null);
    }
    
    public int get(int key) {
        int hashNumber = generateHashKey(key);
        Node current = hashMap[hashNumber];

        while(current.next != null) {
            if (current.next.key == key) {
                return current.next.value;
            }
            current = current.next;
        }

        return -1;
    }
    
    public void remove(int key) {
        int hashNumber = generateHashKey(key);
        Node current = hashMap[hashNumber];

        while(current.next != null) {
            if (current.next.key == key) {
                current.next = current.next.next;
                return;
            }

            current = current.next;
        }
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */